test/irb/test_cmd.rb

```diff
@@ -488,6 +488,47 @@ def show_source_test_method
       assert_empty err
       assert_include(out, code)
     end
+
+    def test_show_source_private_instance
+      pend if RUBY_ENGINE == 'truffleruby'
+      eval(code = <<-EOS, binding, __FILE__, __LINE__ + 1)
+        class PrivateInstanceTest
+          private def show_source_test_method
+            unless true
+            end
+          end unless defined?(show_source_test_method)
```

Comment:
I think this `defined?` is always nil. How about `private_method_defined?`
```ruby
class A
  def foo;end
  def self.bar;end
  p defined?(foo) #=> nil
  p defined?(bar) #=> 'method'
end
```

Reply:
Ah, yes. However, I'm not sure why here's `defined?` is necessary. I added "defined" as in the test case above (`test_show_source_end_finder`). Do you know that?

Reply:
It is first added in this commit https://github.com/ruby/ruby/commit/ca1aa7afeae43b02d00cf2f2a82acde293e48875 and commited to ruby/irb later.
I think it is meant to avoid method-redefinition warning when running the same test case several times.

In ruby/ruby, there are CI that runs test multiple times like `trunk-repeat20-asserts` in `ci.rvm.jp`.
(I didn't know that. ko1 san told me.)

Reply:
Thank you. I fixed it.